File: src/test/java/com/tram/springbootangularboard/security/JwtFactoryTest.java
package com.tram.springbootangularboard.security;

import com.tram.springbootangularboard.config.properties.JwtProperties;
import com.tram.springbootangularboard.domain.Account;
import com.tram.springbootangularboard.domain.UserRole;
import org.junit.Test;
import org.junit.runner.RunWith;
import org.slf4j.Logger;
import org.slf4j.LoggerFactory;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.boot.context.properties.EnableConfigurationProperties;
import org.springframework.boot.test.context.SpringBootTest;
import org.springframework.test.context.junit4.SpringRunner;

import java.util.Arrays;

import static org.junit.Assert.*;

@RunWith(SpringRunner.class)
@SpringBootTest(classes = JwtFactory.class, webEnvironment = SpringBootTest.WebEnvironment.NONE)
@EnableConfigurationProperties(JwtProperties.class)
public class JwtFactoryTest {
    @Autowired
    private JwtFactory jwtFactory;
    private final Logger log = LoggerFactory.getLogger(JwtFactoryTest.class);

    @Test
    public void generateJwtToken() {
        Account account = Account.builder().email("tram").password("password")
                .userRole(Arrays.asList(UserRole.ROLE_USER, UserRole.ROLE_ADMIN)).username("규남").build();
        String jwtToken = jwtFactory.generateJwtToken(AccountContext.fromAccount(account));
        log.info("jwtToken value, {}", jwtToken);
        assertNotNull(jwtToken);
    }
}

File: src/test/java/com/tram/springbootangularboard/security/JwtFactoryTest.kt
package com.tram.springbootangularboard.security

import com.tram.springbootangularboard.domain.Account
import com.tram.springbootangularboard.enum.UserRole
import org.junit.Test
import org.junit.runner.RunWith

import org.slf4j.LoggerFactory
import org.springframework.beans.factory.annotation.Autowired
import org.springframework.boot.test.context.SpringBootTest
import org.springframework.test.context.junit4.SpringRunner

@SpringBootTest
@RunWith(SpringRunner::class)
class JwtFactoryTest {
    private val logger: org.slf4j.Logger = LoggerFactory.getLogger("logger")
    @Autowired
    lateinit var jwtFactory: JwtFactory

    @Test
    fun generateToken() {
        logger.info("hello")
        logger.info("jwt val,{}", jwtFactory.generateToken(AccountContext.fromAccountModel(Account())))
    }

//    companion object {
//        @Deployment
//        fun createDeployment(): JavaArchive {
//            return ShrinkWrap.create(JavaArchive::class.java)
//                    .addClass(JwtFactory::class.java)
//                    .addAsManifestResource(EmptyAsset.INSTANCE, "beans.xml")
//        }
//    }
}
